lib/plausible/data_migration/locations_sync.ex

```diff
@@ -0,0 +1,123 @@
+defmodule Plausible.DataMigration.LocationsSync do
+  @moduledoc """
+  ClickHouse locations data migration for storing location names in clickhouse.
+
+  Run regularly as plausible/locations data changes.
+
+  SQL files available at: priv/data_migrations/LocationsSync/sql
+  """
+  alias Plausible.ClickhouseLocationData
+
+  use Plausible.DataMigration, dir: "LocationsSync", repo: Plausible.IngestRepo
+
+  @columns [
+    %{
+      table: "events_v2",
+      column_name: "country_name",
+      type: "country",
+      input_column: "country_code"
+    },
+    %{
+      table: "events_v2",
+      column_name: "region_name",
+      type: "subdivision",
+      input_column: "subdivision1_code"
+    },
+    %{
+      table: "events_v2",
+      column_name: "city_name",
+      type: "city",
+      input_column: "city_geoname_id"
+    },
+    %{
+      table: "sessions_v2",
+      column_name: "country_name",
+      type: "country",
+      input_column: "country_code"
+    },
+    %{
+      table: "sessions_v2",
+      column_name: "region_name",
+      type: "subdivision",
+      input_column: "subdivision1_code"
+    },
+    %{
+      table: "sessions_v2",
+      column_name: "city_name",
+      type: "city",
+      input_column: "city_geoname_id"
+    },
+    %{
+      table: "imported_locations",
+      column_name: "country_name",
+      type: "country",
+      input_column: "country"
+    },
+    %{
+      table: "imported_locations",
+      column_name: "region_name",
+      type: "subdivision",
+      input_column: "region"
+    },
+    %{
+      table: "imported_locations",
+      column_name: "city_name",
+      type: "city",
+      input_column: "city"
+    }
+  ]
+
+  def out_of_date?() do
+    case run_sql("get-location-data-table-comment") do
+      {:ok, %{rows: [[stored_version]]}} -> stored_version != Location.version()
+      _ -> true
+    end
+  end
+
+  def run() do
+    cluster? = Plausible.MigrationUtils.clustered_table?("sessions_v2")
+
+    {:ok, _} = run_sql("truncate-location-data-table", cluster?: cluster?)
+    {:ok, _} = run_sql("create-location-data-table", cluster?: cluster?)
+
+    countries =
+      Location.Country.all()
+      |> Enum.map(fn country -> %{type: "country", id: country.alpha_2, name: country.name} end)
```

Comment:
here and below I think it's worth pattern matching on `Location` structs, e.g. 

```suggestion
      |> Enum.map(fn %Location.Country{} = country -> %{type: "country", id: country.alpha_2, name: country.name} end
```